cmd/local-gadget/containers/containers.go

```diff
@@ -44,33 +48,69 @@ func NewListContainersCmd() *cobra.Command {
 				return commonutils.WrapInErrParserCreate(err)
 			}
 
-			containers := localGadgetManager.GetContainersBySelector(&containercollection.ContainerSelector{
-				Name: commonFlags.Containername,
-			})
+			containers := localGadgetManager.ContainerCollection.Subscribe(
+				"watch-list-containers",
+				containercollection.ContainerSelector{
+					Name: commonFlags.Containername,
+				},
+				func(event containercollection.PubSubEvent) {
+					// skip if the container isn't enriched
+					if event.Container.ID == "" || event.Container.Name == "" || event.Container.Podname == "" {
+						return
+					}
+					switch event.Type {
+					case containercollection.EventTypeAddContainer:
+						if err = printContainers(parser, commonFlags, []*containercollection.Container{event.Container}, false); err != nil {
+							fmt.Printf("Warning: %s\n", err)
+						}
+					}
+				})
 
 			parser.Sort(containers, []string{"runtime", "name"})
-
-			switch commonFlags.OutputMode {
-			case commonutils.OutputModeJSON:
-				b, err := json.MarshalIndent(containers, "", "  ")
-				if err != nil {
-					return commonutils.WrapInErrMarshalOutput(err)
-				}
-
-				fmt.Printf("%s\n", b)
-			case commonutils.OutputModeColumns:
-				fallthrough
-			case commonutils.OutputModeCustomColumns:
-				fmt.Println(parser.TransformIntoTable(containers))
-			default:
-				return commonutils.WrapInErrOutputModeNotSupported(commonFlags.OutputMode)
+			if err = printContainers(parser, commonFlags, containers, true); err != nil {
+				return err
+			}
+			if optionWatch {
+				stop := make(chan os.Signal, 1)
+				signal.Notify(stop, syscall.SIGINT)
+				<-stop
 			}
-
 			return nil
 		},
 	}
 
+	cmd.Flags().BoolVarP(
+		&optionWatch,
+		"watch", "w",
+		false,
+		"After listing the containers, watch for new containers")
```

Comment:
Please add a new test in `integration/local-gadget/list_containers_test.go` to verify this feature.